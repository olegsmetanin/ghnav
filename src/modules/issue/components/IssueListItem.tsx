import * as React from 'react'

import { WithStyles, createStyles, withStyles } from '@material-ui/core/styles'

import { IIssue } from 'interfaces/issue'
import Link from 'next/link'
import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'

const styles = theme =>
  createStyles({
    paper: {
      padding: theme.spacing.unit * 2
    }
  })

interface IBaseIssueListItemProps {
  value: IIssue
  owner: string
  repo: string
}

export class BaseIssueListItem extends React.Component<IBaseIssueListItemProps & WithStyles<typeof styles>, {}> {
  render() {
    const { classes, value, owner, repo } = this.props
    const { title, number: num, create_at } = value

    return (
      <div>
        <Paper className={classes.paper}>
          <div>
            <Link href={{ pathname: '/issue', query: { owner, repo, number: num } }}>
              <a>
                <Typography variant="h6" color="inherit">
                  #{num} {title} {create_at}
                </Typography>
              </a>
            </Link>
          </div>
          <div />
        </Paper>
      </div>
    )
  }
}

export const IssueListItem = withStyles(styles)(BaseIssueListItem)
