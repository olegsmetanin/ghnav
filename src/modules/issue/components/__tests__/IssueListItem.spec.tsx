/* eslint-env jest */
import { IssueListItem } from '../IssueListItem'
import React from 'react'
import renderer from 'react-test-renderer'

describe('IssueListItem', () => {
  it('renders correct', async () => {
    const props = {
      value: {
        title: 'title',
        owner: 'owner',
        repo: 'repo',
        number: '123',
        create_at: '2018-11-17'
      }
    }
    const component = renderer.create(<IssueListItem {...props} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
