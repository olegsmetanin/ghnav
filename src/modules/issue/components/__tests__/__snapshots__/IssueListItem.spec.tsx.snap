// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueListItem renders correct 1`] = `
<div>
  <div
    className="MuiPaper-root-2 MuiPaper-elevation2-6 MuiPaper-rounded-3 BaseIssueListItem-paper-1"
  >
    <div>
      <a
        href="/issue?owner=&repo=&number=123"
        onClick={[Function]}
      >
        <h6
          className="MuiTypography-root-29 MuiTypography-h6-46 MuiTypography-colorInherit-58"
        >
          #
          123
           
          title
           
          2018-11-17
        </h6>
      </a>
    </div>
    <div />
  </div>
</div>
`;
